CRAMv3.tex

```diff
@@ -920,143 +922,363 @@ \section{\textbf{Record structure}}
+Note however some of these flags can be derived during decode, so may be omitted in the CRAM file and the bits computed based on both reads of a pair-end library residing within the same slice.
+
+\begin{threeparttable}[t]
+\begin{tabular}{|>{\raggedright}p{144pt}|>{\raggedright}p{144pt}|>{\raggedright}p{144pt}|}
+\hline
+\textbf{Bit flag} & \textbf{Comment} & \textbf{Description}\tabularnewline
 \hline
- & \textbf{Data series type} & \textbf{Data series name} & \textbf{Field} & \textbf{Description}\tabularnewline
+0x1 &  & template having multiple segments in sequencing\tabularnewline
 \hline
-1 & int & BF & CRAM bit flags & see CRAM record bit flags\tabularnewline
+0x2 &  & each segment properly aligned according to the aligner\tabularnewline
 \hline
-2 & int & CF  & compression bit flags & see compression bit flags\tabularnewline
+0x4 &  & segment unmapped\tnote{a}\tabularnewline
 \hline
-3 & int & RI & ref id & reference sequence id, not used for single reference slices, 
-reserved for future multiref slices. \tabularnewline
+0x8 & calculated\tnote{b}\ \ or stored in the mate's info & next segment in template unmapped\tabularnewline
 \hline
-4 & int & RL & read length & the length of the read\tabularnewline
+0x10 &  & SEQ being reverse complemented\tabularnewline
 \hline
-5 & int & AP & alignment start & the alignment start position *1\tabularnewline
+0x20 & calculated\tnote{b}\ \ or stored in the mate's info & SEQ of the next segment in the
+template being reversed\tabularnewline
 \hline
-6 & int & RG & read group & the read group identifier\tabularnewline
+0x40 &  & the first segment in the template\tnote{c}\tabularnewline
 \hline
-7 & byte & QS & quality scores & quality scores are stored depending on the value 
-of the `mapped QS included' field\tabularnewline
+0x80 &  & the last segment in the template\tnote{c}\tabularnewline
 \hline
-8 & byte[ ] & RN & read name & the read names (if preserved)\tabularnewline
+0x100 &  & secondary alignment\tabularnewline
 \hline
-9 & *2 & *2 & mate record & *2 (if not the last record)\tabularnewline
+0x200 &  & not passing quality controls\tabularnewline
 \hline
-10 & int & TL & tag ids & tag ids *3\tabularnewline
+0x400 &  & PCT or optical duplicate\tabularnewline
 \hline
-11 & byte[ ] & - & tag values & tag values *3\tabularnewline
+0x800 &  & Supplementary alignment\tabularnewline
 \hline
 \end{tabular}
+\begin{tablenotes}
+\item[a] Bit 0x4 is the only reliable place to tell whether the read is unmapped.  If 0x4 is set, no assumptions may be made about bits 0x2, 0x100 and 0x800.
+\item[b] For segments within the same slice.
+\item[c] Bits 0x40 and 0x80 reflect the read ordering within each template inherent in the sequencing technology used, which may be independent from the actual mapping orientation.
+If 0x40 and 0x80 are both set, the read is part of a linear template (one where the template sequence is expected to be in a linear order), but it is neither the first nor the last read.
```

Comment:
This is counterintuitive so I want to confirm: "first" = true and "last" = true means it is neither first nor last?

Maybe this would be better represented by a two-bit flag 0xC0 with 4 meaningful values?

Reply:
This comes directly from the SAM specification, which basically has identical text bar a bit of clarification on "linear template".

I don't want to confuse things too much for people who are familiar with SAM and while I agree there is potential for a better way to describe it, I think that ship has sailed.

Reply:
ok